scipy/stats/_continuous_distns.py

```diff
@@ -3582,56 +3583,890 @@ class levy_stable_gen(rv_continuous):
+        phi_3_1 = lambda beta, alpha: phi_3(beta, alpha) if beta > 0 else phi_3(-beta, alpha)
+        phi_5 = interpolate.interp2d(beta_range, alpha_range, nu_zeta_table, kind='linear')
+        phi_5_1 = lambda beta, alpha: phi_5(beta, alpha) if beta > 0 else -phi_5(-beta, alpha)
+        
+        # quantiles
+        p05 = np.percentile(data, 5)
+        p50 = np.percentile(data, 50)
+        p95 = np.percentile(data, 95)
+        p25 = np.percentile(data, 25)
+        p75 = np.percentile(data, 75)
+        
+        nu_alpha = (p95 - p05)/(p75 - p25)
+        nu_beta = (p95 + p05 - 2*p50)/(p95 - p05)
+    
+        if nu_alpha >= 2.439:
+            alpha = np.clip(psi_1(nu_beta, nu_alpha)[0], np.finfo(float).eps, 2.)
+            beta = np.clip(psi_2_1(nu_beta, nu_alpha)[0], -1., 1.)
+        else:
+            alpha = 2.0
+            beta = np.sign(nu_beta)
+        scale = (p75 - p25) / phi_3_1(beta, alpha)[0] 
+        zeta = p50 + scale * phi_5_1(beta, alpha)[0]
+        zeta = (zeta - beta * scale * np.tan(np.pi * alpha/2.) 
+                if alpha == 1. else zeta)
+        location = np.clip(zeta, np.finfo(float).eps, np.inf)
+        return (alpha, beta, location, scale)
+
+
+    def _empirical_characteristic_function(self, data, u):
+        empirical_phi = np.mean(np.exp(1j * data * u))
+        return empirical_phi
+
+    def _fitstart_empirical_characteristic_function(self, data):
+        # Regression 1
+        u_k = np.linspace(0.1, 1., 10)
+        empirical_phi_k = np.array([self._empirical_characteristic_function(
+            data, u) for u in u_k])
+        y_k = np.log(-np.log(np.square(np.abs(empirical_phi_k))))
+        x_k = np.log(u_k)
+        slope, intercept = np.polyfit(x_k, y_k, 1)
+        alpha = slope
+        scale = np.exp((intercept - np.log(2))/alpha)
+        # Regression 2
+        u_l = u_k * 0.5
+        empirical_phi_l = np.array([self._empirical_characteristic_function(
+            data, u) for u in u_l])
+        phi_imag_real = empirical_phi_l.imag/empirical_phi_l.real
+        non_principal_int = np.floor((phi_imag_real + np.pi/2.)/np.pi)
+        z_l = np.arctan(phi_imag_real) + np.pi * non_principal_int
+        u_l_alpha = np.float_power(np.abs(u_l), alpha) * np.sign(u_l)
+        x_l = np.vstack([u_l, u_l_alpha]).T
+        slope_1, slope_2 = np.linalg.lstsq(x_l, z_l)[0]
+        location = slope_1
+        slope_2_divisor = np.float_power(scale, alpha) * np.tan(
+                np.pi * alpha/2.)
+        beta = slope_2/slope_2_divisor
+        return (alpha, beta, location, scale)
+
+
+    def _fitstart(self, data):
+        method = getattr(self, 'fit_method', 'quantile')
+        if method == 'ephi_imag_real phi_imag_real mpirical_characteristic_function':
+            return self._fitstart_quantile(data)
+        else:
+            return self._fitstart_empirical_characteristic_function(data)
+        
+    def _stats(self, alpha, beta):
```

Comment:
this is wrong, mean and variance are often undefined / infinite (mean for `alpha <= 1`, variance only finite if `beta == 2` if I recall correctly).

Reply:
I was going by the documentation here: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.html). It says, "if you can’t compute one of [mu, mu2, g1, g2], return it as None." 

But you're right--for consistency with other distributions, I've made the change to return np.nan.